src/main/groovy/io/saagie/plugin/dataops/tasks/service/TechnologyService.groovy

```diff
@@ -11,200 +11,212 @@ import org.gradle.api.logging.Logging
-    }
-
-    def getUntilFind(listVersionElements, accessVersionProperty, upperCaseVersionV1) {
-        def stringVersionV1 = upperCaseVersionV1.toString().toUpperCase() as String
-        if(!SaagieUtils.isCollectionOrArray(listVersionElements)) {
-            return null
-        }
-
-        def versionV2 = listVersionElements.find {
-            accessVersionProperty? it[accessVersionProperty].toUpperCase().equals(stringVersionV1) : it.toUpperCase().equals(stringVersionV1)
-        }
-
-        if(!versionV2) {
-            versionV2 =  listVersionElements.find {
-                accessVersionProperty ? it[accessVersionProperty].toUpperCase().startsWith(stringVersionV1) : it.toUpperCase().startsWith(stringVersionV1)
-            }
-        }
-
-        if(!versionV2){
-            if(stringVersionV1.toString().lastIndexOf(".") > 0) {
-                def reducedUpperCaseVersionV1 = stringVersionV1.substring(0, stringVersionV1.lastIndexOf("."))
-                if(reducedUpperCaseVersionV1  != stringVersionV1) {
-                    return getUntilFind(listVersionElements, accessVersionProperty, reducedUpperCaseVersionV1)
-                } else if (listVersionElements.size()  > 0) {
-                    return listVersionElements[0]
-                }
-            }
-            return null
-        }
-        return versionV2
-    }
-
-    def convertToNumber(String technologyLabel) {
-        return technologyLabel.findAll( /\d+/ )*.toInteger().join().toInteger()
-    }
+	def technologies
+	Map technologiesVersions = [:]
+	OkHttpClient client
+	def technologiesRequest
+	def technologiesVersionsRequest
+	JsonSlurper slurper
+	static final Logger logger = Logging.getLogger(TechnologyService.class)
+	
+	def init(
+			client,
+			technologiesRequest,
+			technologiesVersionsRequest,
+			slurper
+	) {
+		this.client = client
+		this.slurper = slurper
+		this.technologiesRequest = technologiesRequest
+		this.technologiesVersionsRequest = technologiesVersionsRequest
+	}
+	
+	def getTechnologies() {
+		checkReady()
+		Request technologiesCall = technologiesRequest()
+		if (!technologies) {
+			client.newCall(technologiesCall).execute().withCloseable { response ->
+				handleErrors(response)
+				String responseBody = response.body().string()
+				logger.debug("getTechnologies response $responseBody")
+				def parsedTechnologiesData = slurper.parseText(responseBody)
+				if (!parsedTechnologiesData.data || !parsedTechnologiesData.data.technologies) {
+					throwAndLogError("Something went wrong when getting technologies")
+				}
+				technologies = parsedTechnologiesData.data.technologies
+			}
+		}
+		return technologies
+	}
+	
+	def getTechnologyVersions( String technologyId ) {
+		checkReady()
+		if (!technologiesVersions || !technologiesVersions[technologyId]) {
+			Request technologiesVersionRequestCall = technologiesVersionsRequest(technologyId)
```

Comment:
can you put `;` please ?to separate the code

Reply:
`;` is not needed on groovy :)